lib/all_futures/base.rb

```diff
@@ -3,26 +3,28 @@
 module AllFutures
   class Base < ActiveEntity::Base
     prepend ::AllFutures::Callbacks
-    include ActiveModel::Conversion
+    include ::ActiveModel::Conversion
+    include ::AllFutures::Persist
+    include ::AllFutures::Dirty
 
     def initialize(attributes = {})
-      super do
-        extend(::AllFutures::Persist)
-        extend(::AllFutures::Dirty)
-
-        @id ||= SecureRandom.uuid
-        @redis_key = "#{self.class.name}:#{@id}"
-        @new_record = !Kredis.redis.exists?(@redis_key)
-
-        @destroyed = false
-        @previously_new_record = false
-        @_trigger_destroy_callback = false
```

Comment:
The notes on your PR express some completely reasonable confusion about some "why" decisions.

The first explanation is that, for every relevant method I wanted to support, I went to API Dock and studied how the feature is implemented in ActiveRecord 6.1. I was happy to see that in most cases, the logic was quite similar, but I also found a few places where I simply failed to cover cases that would have bitten us in the future. The more I read, the more I found myself refactoring small bits of All Futures logic so that it more closely resembled the AR implementation.

I suppose that you could say that during the course of this analysis, I went from a clean room reverse engineering of functionality, to cargo culting the values returned via console tests, to potentially pulling more yarn than intended from a family heirloom sweater.

For example, go ahead and click "Show Source" on https://apidock.com/rails/v6.1.3.1/ActiveRecord/Persistence/_update_record and you'll see where the `trigger_xxx_callback` ivars originated. It's literally me just sculpting our API to match the function signatures and returned side effects used in AR.

That all explained, I'd be thrilled to see them declared unnecessary and removed! I'm not fighting to keep them, just explaining how they came to be here so we can decide what's right for us.

Reply:
I should dig into their code to see how they structure their callbacks. Maybe we can later tweak things to use their structure where the callback setup doesn't require prepended wrapper methods